### PR TITLE
Integration tests: first suite

### PR DESCRIPTION
**Pull request by mutantcornholio**
Test setup summary:
* Currently only GitHub webhook is tested (as opposed to Matrix trigger)
* All environment is mocked:
    * GitLab/GitHub HTTP endpoints are overridden and lead to `mockttp` servers
    * GitLab/GitHub interaction via git is also overridden and leads to local git daemons
* Fixtures aren't 100% accurate, but tidying it up now seems premature
* Tests are executed with --runInBand, which simplifies setup greatly:
    * command-bot uses filesystem heavily, so having multiple setups running in parallel seems complex
    * multiprocess setup in jest is also not that easy, as global setup can't share any data with child processes (aside from filesystem, which makes it still doable, but complex too)
* Mockttp servers use https, so to get that out of the way, self-signed certs are generated upon setup and used as external CA for node when starting command-bot
* RSA key that used for probot is generated specifically for these tests and isn't used anywhere (safe to store in repo)

* Bugs fixed along the way:
    * CommandRunner broke output when trying to redact an empty token
    * "Probot has started!" message was fired before the setup is done

### CODE REVIEW

**Review by mordamax — APPROVED** [review]

File: src/global.d.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+declare interface Error {
```
[the remark this answers is not in the document]
Comment (mutantcornholio) [reply src/global.d.ts:1]: Somehow, it isn't. I've stumbled upon some discussions on github about it, but can't find those now.

[the remark this answers is not in the document]
Comment (mordamax) [reply src/global.d.ts:1]: I found this, but it doesn't have code.. code looks like more HTTP related error?

image

[the remark this answers is not in the document]
Comment (mutantcornholio) [reply src/global.d.ts:1]: More precisely, [nodejs error](https://nodejs.org/api/errors.html#errorcode).
But the declared type works both with standard JS errors and errors from Node.

[the remark this answers is not in the document]
Comment (mordamax) [reply src/global.d.ts:1]: hmm weird, ya.. 🤷

**Pull request merged**